src/ripe_rainbow/interactive/drivers.py

```diff
@@ -158,7 +158,12 @@ def scroll_to(self, element, sleep = None):
         # in elements inside the page, such as when using an iframe or
         # overflow scroll, therefore we must rely on Web API Element.scrollIntoView()
         # that allows proper scroll operation into element
-        self.instance.execute_script("arguments[0].scrollIntoView();", element)
+        #
+        # the options are needed to scroll the element to the center of the page,

```

Comment:
this only reduces the problem (it's about the fixed top bar right ?). we need to iterate 1px per 1px in the scroll until it becomes visible to the user (is_displayed)

Reply:
That's the first approach we talked about. The reason I didn't go through with is that `is_displayed` does not mean "this is visible and clickable" but rather:

```
/**
 * Determines whether an element is what a user would call "shown". This means
 * that the element is shown in the viewport of the browser, and only has
 * height and width greater than 0px, and that its visibility is not "hidden"
 * and its display property is not "none".
 * Options and Optgroup elements are treated as special cases: they are
 * considered shown iff they have a enclosing select element that is shown.
 *
 * @param {!Element} elem The element to consider.
 * @param {boolean=} opt_ignoreOpacity Whether to ignore the element's opacity
 *     when determining whether it is shown; defaults to false.
 * @return {boolean} Whether or not the element is visible.
 */
bot.dom.isShown = function(elem, opt_ignoreOpacity) {
```

This means that `is_displayed = true` even if the element is below another one and is not clickable.

Given that there is no way to know if an element is clickable, the only alternative I see is:
- Loop:
    1. Is condition met? (e.g. "Was I redirected?")
    2. If ok, exit
    3. If not, change scroll of first parent with scroll (by Z amount in X and Y)

The problem with this approach is the huge overhead it adds in step **1** since it needs some kind of timeout.

Between the approximation in the PR and this method I kind of prefer the first. How often will the middle of the page be covered? I can only think of modals, which only appear if we explicitly trigger some kind of CTA.

Reply:
We really need to go on a good discussion about this topic. This is a complex issue and we really need to find a "final" and easy-to-use solution. @hugo-gomes @gcandal let's all try to talk "deep" about this next Monday ;).